jupyter_client/discovery.py

```diff
@@ -0,0 +1,80 @@
+from .kernelspec import KernelSpecManager
+from .manager import KernelManager
+
+
+class KernelSpecFinder(object):
+    """Find kernels from installed kernelspec directories.
+    """
+    id = 'spec'
+
+    def __init__(self):
+        self.ksm = KernelSpecManager()
+
+    def find_kernels(self):
+        for name, resdir in self.ksm.find_kernel_specs().items():
+            spec = self.ksm._get_kernel_spec_by_name(name, resdir)
+            yield name, {
+                # TODO: get full language info
+                'language': {'name': spec.language},
+                'display_name': spec.display_name,
+                'argv': spec.argv,
```

Comment:
It certainly is, but the aim of this is not necessarily to expose all the information from the kernelspec file; it's to provide a kernel picker with the information it may need to select a kernel. Maybe env is part of that, though. One of the big parts I haven't yet addressed is what information is required in this dictionary, and what standardised but optional fields we'll provide.